google/cloud/pubsub/internal/tracing_message_callback_test.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/tracing_message_callback.h"
+#include "google/cloud/pubsub/message.h"
+#include "google/cloud/pubsub/options.h"
+#include "google/cloud/pubsub/subscription.h"
+#include "google/cloud/pubsub/testing/mock_message_callback.h"
+#include "google/cloud/options.h"
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+#include "google/cloud/pubsub/testing/mock_exactly_once_ack_handler_impl.h"
+#include "google/cloud/internal/opentelemetry.h"
+#include "google/cloud/testing_util/is_proto_equal.h"
+#include "google/cloud/testing_util/opentelemetry_matchers.h"
+#include "google/cloud/testing_util/status_matchers.h"
```

Comment:
only `<opentelemetry/*>` headers in the `#ifdef` blocks pls

Reply:
Done